frontend/lib/src/components/input/input_field.dart

```diff
@@ -12,38 +13,65 @@ import 'package:flutter_form_builder/flutter_form_builder.dart';
   PARAMETER validator: adds validation to the Text Field, it also provides the widget an error message
   as we are using another package for validators, please read https://pub.dev/packages/form_builder_validators
   ex FormBuilderValidators.email() 
+
+  @param labelText, the label which describes the field.
+
+  @param obscureText, whether to hide the text.
+
+  @param keyboardType, the type of keyboard to use for editing the text.
 */
 class InputField extends StatelessWidget {
-  const InputField(
-      {super.key, required this.name, this.validator,});
+  const InputField({
+    super.key,
+    required this.name,
+    required this.labelText,
+    this.obscureText = false,
+    this.keyboardType,
+    this.validator,
+  });
 
   final String name;
+  final String labelText;
+  final bool obscureText;
+  final TextInputType? keyboardType;
   final String? Function(String?)? validator;
 
   @override
   Widget build(BuildContext context) {
-    return Container(
-      decoration: const BoxDecoration(
-        color: Colors.white,
-        borderRadius: BorderRadius.all(Radius.circular(8)),
-        boxShadow: [
-          BoxShadow(
-            color: Colors.black,
-            blurRadius: 2,
-            spreadRadius: 0.1,
-            offset: Offset(0, 2),
-          ),
-        ],
-      ),
-      child: FormBuilderTextField(
-        name: name,
-        validator: validator,
-        decoration: const InputDecoration(
-          border: OutlineInputBorder(
+    return Column(
+      children: [
+        Label(text: labelText),
+        Container(
+          decoration: const BoxDecoration(
+            color: Colors.white,
             borderRadius: BorderRadius.all(Radius.circular(8)),
+            boxShadow: [
+              BoxShadow(
+                color: Color.fromRGBO(0, 0, 0, 0.25),
+                blurRadius: 2,
+                spreadRadius: 0.1,
+                offset: Offset(0, 2),
+              ),
+            ],
+          ),
+          child: FormBuilderTextField(
+            name: name,
+            validator: validator,
+            obscureText: obscureText,
+            keyboardType: keyboardType,
+            decoration: const InputDecoration(
+              border: OutlineInputBorder(
+                borderRadius: BorderRadius.all(Radius.circular(8)),
+              ),
+              contentPadding: EdgeInsets.symmetric(
+                horizontal: 16,
+                vertical: 13,
+              ),
+            ),
+            style: const TextStyle(fontSize: 16),
```

Comment:
either add to text theme and reference sheet ( if "label" doesnt necessarily fit its purpose )
or use existing labelLarge